### PR TITLE
Oop design principles

### PR DESCRIPTION
**Pull request by Daniel-Cretu**

### CODE REVIEW

File: app/Http/Controllers/CommentController.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        // Implement request validation
+
+        $comment = new Comment();
+        $comment->article_id = $articleId;
+        $comment->author_email = $request['formCommentEmail'];
+        $comment->message = $request['formCommentComment'];
+        $comment->save();
+
+        $logger->logModel($request->user(), $comment);
```
Comment (egorbwork) [inline app/Http/Controllers/CommentController.php:22]: Do you still need this change?

File: app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -12,4 +13,14 @@ public function index($productId)
         $product = Product::with('ProductInfo')->where('id',$productId)->get()->all();
         return view('pages.product',['product' => $product]);
     }
+
+    public function addProduct($productId,Request $request, ModelLogger $logger)
+    {
+        $product = Product::with('ProductInfo')->findOrFail($productId);
+        $logger->logModel($request->user(), $product);
```
Comment (egorbwork) [inline app/Http/Controllers/ProductController.php:20]: And this one too?

**Review by egorbwork — CHANGES_REQUESTED** [review]
I need at least feedback to understand if I should merge this one or not.

**Comment by Daniel-Cretu** [comment]
> I need at least feedback to understand if I should merge this one or not.

This branch was based on deprecated **oop-homework** branch and incorporates all it's changes.
My mistake. I haven't payed enough attention when changing the base branch.

I made another branch and a new PR for this task, but for this one I'm using the **master** branch: https://github.com/Daniel-Cretu/bootcamp-laravel/pull/22

**Pull request merged**